react/features/base/devices/middleware.js

```diff
@@ -28,6 +30,12 @@ MiddlewareRegistry.register(store => next => action => {
     case SET_VIDEO_INPUT_DEVICE:
         APP.UI.emitEvent(UIEvents.VIDEO_DEVICE_CHANGED, action.deviceId);
         break;
+    case SHOW_DEVICE_ERROR:
+        store.dispatch(openDialog(DeviceErrorDialog, {
+            micError: action.micError,
+            cameraError: action.cameraError
+        }));
+        break;
```

Comment:
I can't view the old logic...but I think the middleware was doing openDialog but in other places the actions were doing openDialog. So I couldn't tell what the preference is in regards to the call site of openDialog.